src/spatial_reaction_network.jl

```diff
@@ -0,0 +1,109 @@
+        return new(rs, spatial_reactions, DiGraph(lattice))
+    end
+end
+
+### ODEProblem ###
+# Creates an ODEProblem from a LatticeReactionSystem.
+function DiffEqBase.ODEProblem(lrs::LatticeReactionSystem, u0, tspan,
+                               p = DiffEqBase.NullParameters(), args...;
+                               kwargs...)
+    @unpack rs, spatial_reactions, lattice = lrs       
+                        
+    spatial_params = unique(getfield.(spatial_reactions, :rate))
+    pV_in, pE_in = split_parameters(p, spatial_params)
+    nV, nE = length.([vertices(lattice), edges(lattice)])
+    u_idxs = Dict(reverse.(enumerate(Symbol.(getfield.(states(rs), :f)))))
+    pV_idxes = Dict(reverse.(enumerate(Symbol.(parameters(rs)))))
+    pE_idxes = Dict(reverse.(enumerate(unique(getfield.(spatial_reactions, :rate)))))
+
+    u0 = matrix_form(u0, nV, u_idxs)
+    pV = matrix_form(pV_in, nV, pV_idxes)
+    pE = matrix_form(pE_in, nE, pE_idxes);
+
+    return ODEProblem(build_f(lrs, u_idxs, pE_idxes), u0, tspan, (pV,pE), args...; kwargs...)
+end
+
+# Splits parameters into those for the compartments and those for the connections.
+split_parameters(parameters::Tuple, spatial_params) = parameters
+split_parameters(parameters::Vector, spatial_params) = filter(p -> !in(p[1], spatial_params), parameters), filter(p -> in(p[1], spatial_params), parameters);
+
+# Converts species and parameters to matrices form.
+matrix_form(input::Matrix, args...) = input
+matrix_form(input::Vector{Pair{Symbol,Vector{Float64}}}, n, index_dict) = mapreduce(permutedims, vcat, last.(sort(input, by=i -> index_dict[i[1]])))
+matrix_form(input::Vector, n, index_dict) = matrix_form(map(i -> (i[2] isa Vector) ? i[1] => i[2] : i[1] => fill(i[2], n), input), n, index_dict);
+
+# Creates a function for simulating the spatial ODE with spatial reactions.
+function build_f(lrs::LatticeReactionSystem, u_idxs::Dict{Symbol,Int64}, pE_idxes::Dict{Symbol,Int64})
+    ofunc = ODEFunction(convert(ODESystem, lrs.rs))
+
+    return function internal___spatial___f(du, u, p, t)
+        # Updates for non-spatial reactions.
+        for comp_i in 1:size(u,2)
+            ofunc((@view du[:,comp_i]), (@view u[:,comp_i]), p[1], t)
+        end
+        
+        # Updates for spatial reactions.
+        for comp_i in 1:size(u,2)
+            for comp_j::Int64 in (lrs.lattice.fadjlist::Vector{Vector{Int64}})[comp_i], sr::SpatialReaction in lrs.spatial_reactions::Vector{SpatialReaction}
+                rate = get_rate(sr,p[2],(@view u[:,comp_i]),(@view u[:,comp_j]), u_idxs, pE_idxes)
+                foreach(stoich -> du[u_idxs[stoich[1]],comp_i] += rate*stoich[2], sr.net_stoich_src)
+                foreach(stoich -> du[u_idxs[stoich[1]],comp_j] += rate*stoich[2], sr.net_stoich_dst)
+            end
+        end
+    end
+end
+
+# Get the rate of a specific reaction.
+function get_rate(sr, pE, u_src, u_dst, u_idxs, pE_idxes)

```

Comment:
Why not reuse the current ODE rate generation code for non-spatial `ReactionSystem`s?

Reply:
I had a look at this, but didn't really see how I could reuse it given that here we have two sets of substrates (one for each of the two neighbouring compartments.

Also, for normal ODEs we just generate a symbolic expression with `oderatelaw`, while here we actually have to calculate a number to plug into our ODE function. Similarily I was a bit uncertain how to utilise symbolic rates in the SpatrialReaction, given that we at some point have to go back to just dealing with numbers, and have to get a value from the symbolic expression and the `p` and `u` matrices.